### PR TITLE
Add RPC Endpoint tests using PowerMock

### PR DESCRIPTION
**Pull request by kaladay**
# What does this pull request do?
Add unit tests for the RPC Endpoint using PowerMock.

This demostrates a simple approach to unit testing endpoints.

This test works by confirming that the `action.run()` is called.

# What's new?
* Add PowerMock 2.0.9 test dependencies.

### CODE REVIEW

**Review by wwelling — APPROVED** [review]

File: api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpointTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.Action;
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.OperationResult;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ActionPool.class)
+public class RPCEndpointTest {
+
+	final static String URI_TEST = "/api/rpc/test";
+
+	private static RESTEndpoint restEndpoint;
```
Comment (chenejac) [inline api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpointTest.java:34]: private static RPCEndpoint rpcEndpoint ????

**Review by chenejac — CHANGES_REQUESTED** [review]
@kaladay Thanks for this. You are testing RESTEndpoint with RPCEndpointTest and using URI for RPCEndpoint (/api/rpc/test). Not sure there is any sense in testing doGet, doPost, doPut, and doDelete of RPCEndpoint separately. @litvinovg any chance those endpoint will be different at the end of sprint?

File: api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpointTest.java (hunk 1 above)
Comment (kaladay) [reply api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpointTest.java:34]: Good catch, I'll get right to this.

Comment (kaladay) [reply api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RPCEndpointTest.java]: Fixing this required me to fix many more things.
I practically rewrote the tests and had to merge staging into this branch.
It is now ready for review again.

**Review by wwelling — APPROVED** [review]

**Review by chenejac — APPROVED** [review]

**Review by litvinovg — APPROVED** [review]

**Pull request merged**